resolver-dns/src/main/java/io/netty/resolver/dns/ShuffledDnsServerAddressStream.java

```diff
@@ -58,6 +63,16 @@ public InetSocketAddress next() {
     }
 
     @Override
+    public int size() {
+        return addresses.length;
+    }
+
+    @Override
+    public DnsServerAddressStream clone() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this "clone" implementation; use a copy constructor or copy factory instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2975)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use super.clone() to create and seed the cloned instance to be returned. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1182)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The return type of the clone() method must be the class name when implements Cloneable [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3ACloneMethodReturnTypeMustMatchClassName)